src/app/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,61 @@
+import { useNewsContext } from '@/context/news.context';
+import React from 'react';
+import { Selects } from '@/models/types';
+import classNames from 'classnames/bind';
+import getPaginationRange from '@/utils/getPaginationRange';
+import styles from './Pagination.module.scss';
+
+type Props = {
+  currentPage: number;
+  setCurrentPage(value: number): void;
+  selects: Selects;
+  setSelects(selects: Selects): void;
+};
+
+const cx = classNames.bind(styles);
+
+function Pagination({ currentPage, setCurrentPage, selects, setSelects }: Props) {
+  const { news } = useNewsContext();
+  const totalPageNumber = Math.floor(100 / news.length); // 100 articles - API limit
+
+  const paginationElements = getPaginationRange(totalPageNumber, currentPage);
+
+  const handlePageChange = (value: number) => {
+    if ((value < currentPage && currentPage > 1) || (value > currentPage && currentPage < totalPageNumber)) {
```

Comment:
What do you think of replacing this condition with `setCurrentPage(Math.max(1, Math.min(totalPageNumber, value)));` ?